apis/operator/v1alpha1/apigateway_types.go

```diff
@@ -70,3 +74,12 @@ type APIGatewayList struct {
 func init() {
 	SchemeBuilder.Register(&APIGateway{}, &APIGatewayList{})
 }
+
+// IsInGracefulDeletion returns true if the APIGateway is in deletion process, false otherwise.
+func (a *APIGateway) IsInGracefulDeletion() bool {
```

Comment:
Is `IsInGracefulDeletion` a good name for this? After all we don't have any other deletion strategy, nor does this check ensure that it is actually graceful

Reply:
The deletion is per default graceful, since finalizers can exist. It'S also called like that in the [docs](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1@v0.28.2#ObjectMeta.DeletionTimestamp)
> This field is set by the server when a graceful deletion is requested by the user

But we can also rename it to `IsInDeletion`, because it might be easier to understand.